### PR TITLE
More pytest optimizations

### PR DESCRIPTION
**Pull request by rustyrussell**
1. A flake fix for  test_restart_many_payments.
2. A flake fix for test_onchain_their_unilateral_out
3. Mass conversion to get_nodes() which batches, vs. individual get_node() calls.
4. Turn off valgrind for SLOW_MACHINE slow tests, rather than skipping them entirely.

Changelog-None

### CODE REVIEW

**Comment by rustyrussell** [comment]
Trivial rebase

**Comment by rustyrussell** [comment]
Trivial rebase, and fix to *actually* turn off VALGRIND on slow tests!

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 1]
```diff
@@ -1094,8 +1094,8 @@ def join_nodes(self, nodes, fundchannel=True, fundamount=10**6, wait_for_announc
             bitcoind.rpc.sendtoaddress(addr, (fundamount + 1000000) / 10**8)
 
         bitcoind.generate_block(1)
+        sync_blockheight(bitcoind, nodes)
```
Comment (niftynei) [inline contrib/pyln-testing/pyln/testing/utils.py:1102]: this seems like a different commit?

File: tests/test_plugin.py [hunk 2]
```diff
@@ -774,6 +774,9 @@ def test_channel_opened_notification(node_factory):
     amount = 10**6
     l1, l2 = node_factory.line_graph(2, fundchannel=True, fundamount=amount,
                                      opts=opts)
+
+    # Might have already passed, so reset start.
+    l2.daemon.logsearch_start = 0
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply tests/test_plugin.py:779]: Yep, split that commit out.

File: contrib/pyln-testing/pyln/testing/utils.py (hunk 1 above)
Comment (rustyrussell) [reply contrib/pyln-testing/pyln/testing/utils.py:1102]: I split it, too.

**Comment by cdecker** [comment]
ACK 6c9d196807e5661c54029e2afa6b91037fd23fe3

**Pull request merged**